### ISSUE TITLE
Request: change ITV subtitle files naming

### ISSUE THREAD

**Issue by jefro108** [GetiPlayerAutomator/get-iplayer-automator#566]
I've just noticed that the subtitle files created for ITV programmes have the form (with Kodi naming) of
Programme Title.s01e01.Episode 1.mp4.srt 
with the programme itself as
Programme Title.s01e01.Episode 1.mp4

whereas for a BBC TV programme is is created as
Programme Title.s01e01.Episode 1.srt 
with the programme itself as
Programme Title.s01e01.Episode 1.mp4

The latter works for automatically loading subtitles into a video player or in Kodi.

Please could the .mp4 be removed from the subtitle naming?

**Comment by dinkypumpkin** [comment]
Both forms work in Kodi and VLC. If your media player isn't that capable, patches are welcome.

**Comment by jefro108** [comment]
I'm using 5KPlayer (since it has AirPlay streaming available http://www.5kplayer.com/airplay/) and it only seems to choose the BBC version of the subtitle :(

**Comment by larryy** [comment]
In case the OP is unaware, VLC gives you control over the "fuzziness" of subtitle filename matching.  I think you have to invoke the advanced settings, then look in Video > Subtitles / OSD, where you should see "Subtitle autodetection fuzziness".  Experiment with it to find a value that automatically finds your subtitle files.

**Comment by dinkypumpkin** [comment]
Default VLC fuzziness setting is good enough for this case. As for 5kplayer, you'll have to take up its limitations with its developers. You can always rename the subtitle file or open it directly in the player.

**Comment by larryy** [comment]
If you do any programming, whipping out a Python script to crawl nested directories looking for *.mp4.srt files and renaming them to *.srt files would be pretty easy.  And with Platypus you can convert the script to a stand-alone Mac app that accepts drag and drop.  I can't promise to do it in a day, but in a week I should be able to steal time to knock one out if you want it.

**Comment by ndcoxhead** [comment]
But to fix it in GIA (if it is unintentional i.e. a bug) should be a one line fix, rather than hacking the output files. I think the fix would have taken less time than all these conversations.

**Comment by larryy** [comment]
Well, dinkypumpkin did say he'd accept pull requests.

I've never been comfortable with objective-c, which is what GiA is written in.  I did take a quick look, and in ITVDownloader.m I see where he handles the subtitle URL, but I don't see where he handles the filename.  In BBCDownloader.m I see where he handles the filename.  Dunno, maybe that code does double duty and gets invoked for ITV as well.  If it was C++ I'd try tweaking srtPath -- remove ".mp4" it it's at the end of the string -- and see what happens.

It's also been ages since I used xCode.  Just for giggles I tried building GiA.  The build fails attempting to run a build script launching /bin/sh.  No idea why.  Ah well.  I don't have time for this now.

So, no, not quicker.

(I have nothing to do with GiA development.  I'm just a happy, grateful user.  I do lots of other programming, but am not familiar enough with the tools used for GiA development to help quickly, and don't have all that much spare time.  I'm sure dinkypumpkin would be happy to have your help, if you really think it's so easy.)
